crates/bevy_ecs/src/world/entity_ref.rs

```diff
@@ -564,9 +564,22 @@ impl<'w> EntityMut<'w> {
     /// # assert_eq!(new_r.0, 1);
     /// ```
     pub fn world_scope<U>(&mut self, f: impl FnOnce(&mut World) -> U) -> U {
-        let val = f(self.world);
-        self.update_location();
-        val
+        struct Guard<'w, 'a> {
+            entity_mut: &'a mut EntityMut<'w>,
+        }
+
+        impl Drop for Guard<'_, '_> {
+            #[inline]
+            fn drop(&mut self) {
+                self.entity_mut.update_location();
```

Comment:
Can we document that `world_scope()` calls `update_location()`? This was missing in the original code too.

Reply:
This was a conscious choice not to mention unsafe implementation details in the docs of a safe method

Reply:
Fair. But on the other hand if I see the `update_location()` on docs.rs then I will immediately ask myself "should I use that after `world_scope()`, and safety doesn't answer my question; if anything I'd be tempted to think I _do_ need it since it's unsafe but `world_scope()` is not.

Reply:
The docs for `EntityMut::update_location` say that it only needs to be called if you are using `EntityMut::world_mut`. Perhaps it could be phrased differently to make that more clear?

Reply:
Then that should be fine.